hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadata.java

```diff
@@ -170,7 +170,8 @@ Map<Pair<String, String>, HoodieMetadataColumnStats> getColumnStats(final List<P
    * @return {@link HoodieData} of {@link HoodieRecord}s with records matching the passed in key prefixes.
    */
   HoodieData<HoodieRecord<HoodieMetadataPayload>> getRecordsByKeyPrefixes(List<String> keyPrefixes,
-                                                                          String partitionName);
+                                                                          String partitionName,
+                                                                          boolean shouldLoadInMemory);
```

Comment:
can we declare a member variable named `shouldLoadColumnStatsInMemory` in `FileSystemBackedTableMetadata`, so that we don't need to change the definition of `getRecordsByKeyPrefixes`.

Reply:
Good call. I think that would definitely make sense if we in consider in the future to make all methods be configurable to either load in-memory or using Spark. Right now only this method is configurable and it would be very misleading if we elevate it to be instance level config